app/views/data_view.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+from app.models import Donor, Donation, Item
+from django.http import HttpResponse, JsonResponse, HttpResponseBadRequest
+from django.views import View
+import simplejson as json
+
+@login_required(login_url='/login')
+def aggregate_value(request):
+    '''if request.GET['interval'] is false, return value of all items
+    otherwise, return array of values for given interval
+    '''
+    try:
+        start_date = request.GET['start_date']
+        end_date = request.GET['end_date']
+
+        items = __getModelsGivenInterval('item', start_date, end_date)
```

Comment:
Objects should be retrieved based on model param in the request.GET rather than being hard coded into 'item'

Reply:
I thought aggregate_value's return value doesn't depend on the model (only care about item), so I didn't expect to have model in GET request. Am I getting wrong?

Reply:
Ok yea. I guess that's fine for now. Might need to change in the future though. 👍